### DISCUSSION TITLE
Questions about the prediction of multiple time series

### DISCUSSION THREAD

**Discussion by yeungsk** [Nixtla/mlforecast#130] [Q&A] [answered]
Hi, I have a question regarding the prediction of multiple time series. When I specify the id_col in  `MLForecast.fit`, does the model train one series one time and combine the result? or does it train based on all the series(cross-learning)?

**Comment by jmoralez** [comment]
The id_col is used to compute features per serie, but the model is trained using all series.

**Reply by yeungsk** [reply]
Thanks. Because some models don't support category features, how did you handle the `id_col` when training the data? Did you use an encoding with the `id_col`?

**Reply by jmoralez** [reply, marked as answer]
The id_col is used only as an identifier, it isn't used as a feature. If you want to use it as a feature you have to create an extra column and pass it to `static_features` as is done [here](https://www.kaggle.com/code/lemuz90/m5-mlforecast-eval?scriptVersionId=127391281&cellId=15). If your model doesn't support categorical features you'll have to encode it.

**Reply by yeungsk** [reply]
Thank you!

**Comment by edgBR** [comment]
Hi,

Then it means that the training is performed as follows (Dependend Multivariate forecasting):

![image](https://github.com/Nixtla/mlforecast/assets/14976422/1f680ae2-a221-452d-8e7d-b3e329595532)
![image](https://github.com/Nixtla/mlforecast/assets/14976422/9d2af7e0-c4fc-4d26-816f-d8b0047f700f)

Or as follows (Independent Multi Series Forecasting):

![image](https://github.com/Nixtla/mlforecast/assets/14976422/97da16ea-6128-48b8-a2ea-35a1d04cf75d)
![image](https://github.com/Nixtla/mlforecast/assets/14976422/0a03a8f2-a750-452b-a729-c2dead275500)

Im asking this because of course if you use MLP or any linear regression model, dependent training is possible. But if you are using XGBoost how you are predicting multiple Y1,Y2,Y3 if unique id is not used as a feature?

**Reply by jmoralez** [reply]
Hi. It's the second approach

**Reply by atsangarides** [reply]
Apologies for responding to an old threat. Does that mean, that if I have series s1 and s2, if I train one model per series and if I train a single model for both, assuming features and range are exactly the same (i.e. just a concatenation of dataframes for the second approach) would the predictions be identical, or the two approaches would yield different results? I.e. are you training 2 models under the hood in the second approach of a single one?

**Reply by jmoralez** [reply]
It's a single model
